eclipse-collections/src/main/java/org/eclipse/collections/impl/map/sorted/mutable/SynchronizedSortedMap.java

```diff
@@ -122,6 +123,16 @@ public SynchronizedSortedMap(MutableSortedMap<K, V> newMap, Object lock)
         }
     }
 
+    @Override
+    public MutableSortedMap<K, V> withMap(Map<? extends K, ? extends V> map)
+    {
+        synchronized (this.lock)
+        {
+            this.putAll(map);
+            return this;
```

Comment:
Some of the synchronized wrappers call this.putAll and some call this.getDelegate.putAll. I prefer this.putAll, like we see here.